src/services/auth.service.ts

```diff
@@ -0,0 +1,37 @@
+import axios, { getAuthUrl } from "../api/index";
+import { ITokens } from "../store/reducers/user/user.interface";
+import { IRegisterResponse } from "../store/reducers/auth/auth.interface";
+import localstorageService from "./localstorage.service";
+
+class AuthService {
+  async login(username: string, password: string) {
+    const response = await axios.post<ITokens>(getAuthUrl("login"), {
+      username,
+      password,
+    });
+
+    if (response.data.accessToken) {
+      localstorageService.set("accessToken", response.data.accessToken);
+      localstorageService.set("refreshToken", response.data.refreshToken);
```

Comment:
If you are using manually localStorage without redux-persist, will be better to create a service for that.

Example:
`const ACCESS_TOKEN_KEY = 'token';

export const localStorageAuthService = (() => {
	// TOKEN
	function privateSetAccessToken(token: string): void {
		localStorage.setItem(ACCESS_TOKEN_KEY, token);
	}

	function privateGetAccessToken(): string | null {
		return localStorage.getItem(ACCESS_TOKEN_KEY);
	}

	function privateClearAccessToken(): void {
		localStorage.removeItem(ACCESS_TOKEN_KEY);
	}

	// CLEAR STORAGE
	function privateClearStorage(): void {
		localStorage.clear();
	}

	return {
		// TOKEN
		setAccessToken: privateSetAccessToken,
		getAccessToken: privateGetAccessToken,
		clearAccessToken: privateClearAccessToken,
		// CLEAR STORAGE
		clearStorage: privateClearStorage,
	};
})();`

Reply:
This can be used for all cases where you are using a localStorage call.